propolis/src/hw/nvme/ns.rs

```diff
@@ -0,0 +1,226 @@
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex};
+
+use crate::block::*;
+use crate::hw::nvme::bits::RawCompletion;
+use crate::hw::nvme::cmds::{self, NvmCmd};
+use crate::{common, dispatch::DispCtx};
+
+use super::bits::{self, RawSubmission};
+use super::cmds::{Completion, ReadCmd, WriteCmd};
+use super::queue::{CompQueue, SubQueue};
+use super::NvmeError;
+
+const BLOCK_SZ: u64 = 512;
+
+pub struct NvmeNs {
+    pub ident: bits::IdentifyNamespace,
+    bdev: Arc<dyn BlockDev<Request>>,
+}
+
+impl NvmeNs {
+    pub fn create(bdev: Arc<dyn BlockDev<Request>>) -> Self {
+        let binfo = bdev.inquire();
+        let total_bytes = binfo.total_size * binfo.block_size as u64;
+        let nsze = total_bytes / BLOCK_SZ;
```

Comment:
Is there something that constraints this such that binfo.block_size is always a multiple of BLOCK_SZ?

Reply:
In theory, I guess not but in practise our current `PlainBdev` implementation defines its block size as 512.